client/state/jetpack/credentials/reducer.js

```diff
@@ -63,6 +66,21 @@ export const updateRequestStatus = keyedReducer( 'siteId', ( state, { type } ) =
 	return state;
 } );
 
+export const testRequestStatus = keyedReducer( 'siteId', ( state, { type, role, testResult } ) => {
+	switch ( type ) {
+		case JETPACK_CREDENTIALS_TEST:
+			return { [ role ]: 'pending' };
```

Comment:
Won't this override any other existing keys for other roles in this reducer? Should we maintain them?

Reply:
I think this is okay. As soon as we start the credentials check process, this state will be `pending` until we get a response, then it will be `valid` or `invalid`.

Reply:
In that case, since we always override the state for a site, I'd recommend a different schema, though, because the one suggested here is less predictable.

How about something like:

```
{ role: 'main', status: 'invalid' }
```

Reply:
As we discussed today, we missed to keep existing keys on that state, so we are now adding:

```
case JETPACK_CREDENTIALS_TEST_VALID:
	return {
		...state,
		[ role ]: 'valid',
	};
```